### PR TITLE
Request Screen + Buttons + Logos  Enhancements  

### PR DESCRIPTION
**Pull request by aoutz**
## **Overview**  
This PR introduces several UI enhancements to improve the app's visual consistency, usability, and branding. Key improvements include adding a background to the Create Request screen, harmonizing button styles across the app, integrating service-specific icons on the Provider Info page, and adding the app logo to key screens like Inbox, Orders, and Profile.

---

### **1. Create Request Screen:**  
- 🖼️ **Added a visually appealing background image** covering the entire screen.  
- 📐 **Adjusted padding and layout elements** for a better user experience.  

### **2. Button Harmonization:**  
- 🎨 **Standardized button styles** (height, width, corner radius, colors).  
- 🔧 **Ensured consistent button padding, icons, and text alignment.**  

### **3. Provider Info Screen:**  
- 🏷️ **Displayed service-specific icons** based on the provider's available services.  
- 📊 **Improved service detail layout** with proper spacing and alignment.  

### **4. App Logo Integration:**  
- 🖋️ **Added the app logo** to:  
  - 📥 **Inbox Screen**  
  - 📦 **Orders Screen**  
  - 👤 **Profile Screen**  
- 📐 **Adjusted logo positioning** to maintain a clean and professional look.  

---

## **Why This Matters**  
- **🎯 Improved Usability:** Consistent buttons enhance user interaction.  
- **✨ Enhanced Visual Appeal:** Backgrounds and icons add personality and context.  
- **🆔 Brand Consistency:** Placing the app logo on major screens strengthens brand identity.  

---

## **Testing & Verification**  
- ✅ Verified layout and appearance across various devices and screen sizes.  
- 🧪 Tested button clicks, icon visibility, and app logo positioning.  
- 🔄 Adjusted tests to account for new actions, including the **Help & Support** and **Read More on GitHub** buttons.  

- 🚫 Ensured no functionality regressions after applying the UI updates.  

---

**This PR is ready for review.** Feedback and suggestions are highly appreciated! 😊

### CODE REVIEW

File: app/src/main/java/com/android/solvit/seeker/ui/provider/ProviderInfo.kt
Comment (hjamaly) [inline app/src/main/java/com/android/solvit/seeker/ui/provider/ProviderInfo.kt:1]: 👍 good add of service icon

File: app/src/main/java/com/android/solvit/seeker/ui/request/CreateRequest.kt
Comment (hjamaly) [inline app/src/main/java/com/android/solvit/seeker/ui/request/CreateRequest.kt:1]: Wow ! once again great work on implementing a clean screen

File: app/src/main/java/com/android/solvit/shared/ui/utils/TopAppBar.kt
Comment (hjamaly) [inline app/src/main/java/com/android/solvit/shared/ui/utils/TopAppBar.kt:1]: Be careful with background color with dark mode we'll have white as background ! We'll fix it in dark mode consistency pr

**Review by hjamaly — APPROVED** [review]
Great adds with a beautiful screen implemented to create a request that makes the app more consistent with other screens ! Thank you, be careful with dark mode consistency on some modifs

File: app/src/main/java/com/android/solvit/seeker/ui/profile/SeekerProfile.kt
Comment (manunam2002) [inline app/src/main/java/com/android/solvit/seeker/ui/profile/SeekerProfile.kt:1]: nice additions !!

File: app/src/main/java/com/android/solvit/seeker/ui/request/CreateRequest.kt
Comment (manunam2002) [inline app/src/main/java/com/android/solvit/seeker/ui/request/CreateRequest.kt:1]: Nice screen, mind that it doesn't change in dark mode, but don't worry I will fix this in the dark mode pr

File: app/src/main/java/com/android/solvit/seeker/ui/service/Services.kt
Comment (manunam2002) [inline app/src/main/java/com/android/solvit/seeker/ui/service/Services.kt:1]: This is okay but you should change it as well in the Request Feed

File: app/src/main/res/drawable/discount_bg.png
Comment (manunam2002) [inline app/src/main/res/drawable/discount_bg.png:1]: why did you change this? now it is not centered (at least on my phone)

**Review by manunam2002 — APPROVED** [review]
### Feedback 

This PR revisits the Request Screen, which is indeed one of the most important screens of our app, and finally makes it good looking. Please be careful about the fact that changes work across different devices, and dark mode is taken into account (for this one you can not worry about it since I have a branch that fixes this), but overall, great addition and great job!

**Review by achillet3 — APPROVED** [review]
# PR #486 Review: Request Screen + Buttons + Logos Enhancements

## Overview
This PR introduces significant UI/UX improvements focusing on visual consistency, branding, and user experience. The changes span across multiple screens and components, enhancing the overall app quality.

## Strengths 🌟

### Enhanced Visual Identity
- Added service-specific icons to provider profiles, improving visual context
- Integrated app logo across key screens (Inbox, Orders, Profile)
- Consistent use of background images and styling
- Added beautiful background image to Create Request screen with proper ContentScale.Crop

### Improved Button Design
- Standardized button styles with consistent height, width, and corner radius (25.dp)
- Better visual hierarchy in the UI
- Consistent text casing ("Book now" instead of "Book Now")
- Enhanced button feedback with proper color schemes for enabled/disabled states

### Better User Experience
- Added helpful links to GitHub documentation in profile section
- Improved profile image presentation with service icon overlay
- Enhanced scrolling behavior in provider registration
- Added AI Assistant dialog with image analysis capabilities
- Improved form validation and feedback

### Code Quality
- Well-structured Composable functions with clear documentation
- Consistent use of test tags for UI testing
- Good separation of concerns in UI components
- Proper state management with remember and mutableState
- Improved error handling in image selection

## Issues to Address 🔧

### 1. Test Coverage
```kotlin
// SeekerProfileTest.kt
// Changed from performClick() to assertHasClickAction()
// This reduces test reliability as it only checks if clicking is possible,
// not if the click actually works
fun profileOptionItemsAreClickable() {
  composeTestRule.onNodeWithTag("MyAccountOption").assertHasClickAction()
  // ...
}
```
**Recommendation**: Keep the `performClick()` tests and add navigation assertions to ensure full functionality.

### 2. Error Handling
```kotlin
// SeekerProfile.kt
fun openWebPage(context: Context, url: String) {
  try {
    val webIntent = Intent(Intent.ACTION_VIEW, Uri.parse(url))
    // ...
  } catch (e: Exception) {
    Toast.makeText(context, "Unable to open link", Toast.LENGTH_SHORT).show()
  }
}
```
**Issues**:
- Generic error message
- No error logging
- No specific exception handling

### 3. UI Consistency
```kotlin
// ProviderInfo.kt
Box(
    modifier = Modifier
        .padding(4.dp)
        .size(30.dp)
        .clip(RoundedCornerShape(8.dp))
)
```
**Issue**: Inconsistent corner radius values (8.dp vs 16.dp/25.dp used elsewhere)

### 4. Performance Considerations
- Image loading in `RequestScreen.kt` could benefit from Coil's memory caching
- Multiple state updates in image selection could be optimized
- Consider lazy loading for service icons

### 5. Dark Mode and Readability Issues
In `RequestScreen.kt`, the background image implementation has several issues:
```kotlin
Box(modifier = Modifier.fillMaxSize()) {
    // Background Image
    Image(
        painter = painterResource(id = R.drawable.bg_request),
        contentDescription = null,
        contentScale = ContentScale.Crop,
        modifier = Modifier.fillMaxSize().testTag("requestBackground"))

    Scaffold(
        modifier = Modifier.padding(16.dp).testTag("requestScreen"),
        containerColor = Color.Transparent,
        // ...
    )
}
```
**Issues**:
- No dark mode support - the background image doesn't adapt to system theme
- Poor readability in light mode - text contrast might be insufficient depending on the background image
- Missing fallback background color for accessibility
- No content description for screen readers

**Recommendations**:
1. Add theme-aware background:
```kotlin
@Composable
fun RequestScreen(/*...*/) {
    Box(modifier = Modifier.fillMaxSize()) {
        // Background Image with overlay
        Box {
            Image(
                painter = painterResource(id = R.drawable.bg_request),
                contentDescription = "Request form background image",
                contentScale = ContentScale.Crop,
                modifier = Modifier.fillMaxSize()
            )
            Box(
                modifier = Modifier
                    .fillMaxSize()
                    .background(
                        colorScheme.surface.copy(
                            alpha = if (isSystemInDarkTheme()) 0.8f else 0.4f
                        )
                    )
            )
        }

        Scaffold(
            modifier = Modifier.padding(16.dp).testTag("requestScreen"),
            containerColor = Color.Transparent,
            // ...
        )
    }
}
```

2. Improve form field readability:
```kotlin
@Composable
fun TitleInput(title: String, onTitleChange: (String) -> Unit) {
    OutlinedTextField(
        value = title,
        onValueChange = onTitleChange,
        label = { Text("Title", style = Typography.bodyLarge) },
        placeholder = { Text("Name your Request", style = Typography.bodyLarge) },
        shape = RoundedCornerShape(12.dp),
        modifier = Modifier.fillMaxWidth().testTag("inputRequestTitle"),
        colors = OutlinedTextFieldDefaults.colors(
            unfocusedContainerColor = colorScheme.surface.copy(alpha = 0.7f),
            focusedContainerColor = colorScheme.surface.copy(alpha = 0.9f),
            focusedBorderColor = colorScheme.primary,
            unfocusedBorderColor = colorScheme.outline,
            focusedTextColor = colorScheme.onSurface,
            unfocusedTextColor = colorScheme.onSurface
        )
    )
}
```

3. Add a fallback for failed image loading:
```kotlin
Box(modifier = Modifier.fillMaxSize()) {
    Box(
        modifier = Modifier
            .fillMaxSize()
            .background(colorScheme.surface)  // Fallback color
    )
    Image(
        painter = painterResource(id = R.drawable.bg_request),
        contentDescription = "Request form background image",
        contentScale = ContentScale.Crop,
        modifier = Modifier.fillMaxSize(),
        alpha = if (isSystemInDarkTheme()) 0.6f else 0.8f
    )
    // Rest of the UI
}
```

These changes will:
- Provide better contrast in both light and dark modes
- Improve text readability across different backgrounds
- Add proper accessibility support
- Handle image loading failures gracefully
- Maintain visual consistency with the app's theme

## Suggestions for Improvement 💡

### Component Reusability
```kotlin
@Composable
fun ServiceIconOverlay(
    service: Services,
    modifier: Modifier = Modifier
) {
    Box(
        modifier = modifier
            .padding(4.dp)
            .size(30.dp)
            .clip(RoundedCornerShape(16.dp))  // Standardized corner radius
            .background(OnPrimary),
        contentAlignment = Alignment.Center
    ) {
        Icon(
            painter = painterResource(id = Services.getIcon(service)),
            contentDescription = "Service: ${Services.format(service)}",
            tint = Color.Unspecified,
            modifier = Modifier.size(25.dp)
        )
    }
}
```

### Improved Error Handling
```kotlin
sealed class WebPageError {
    object NetworkError : WebPageError()
    object InvalidUrl : WebPageError()
    object ActivityNotFound : WebPageError()
}

fun openWebPage(context: Context, url: String): Result<Unit> {
    return try {
        val webIntent = Intent(Intent.ACTION_VIEW, Uri.parse(url)).apply { 
            flags = Intent.FLAG_ACTIVITY_NEW_TASK 
        }
        context.startActivity(webIntent)
        Result.success(Unit)
    } catch (e: ActivityNotFoundException) {
        Log.e("WebPage", "No browser found", e)
        Toast.makeText(context, "Please install a web browser", Toast.LENGTH_SHORT).show()
        Result.failure(e)
    } catch (e: Exception) {
        Log.e("WebPage", "Failed to open link", e)
        Toast.makeText(context, "Unable to open link: ${e.localizedMessage}", Toast.LENGTH_SHORT).show()
        Result.failure(e)
    }
}
```

### Image Loading Optimization
```kotlin
@Composable
fun OptimizedImageLoader(
    imageUrl: String?,
    contentDescription: String?,
    modifier: Modifier = Modifier
) {
    AsyncImage(
        model = ImageRequest.Builder(LocalContext.current)
            .data(imageUrl)
            .crossfade(true)
            .placeholder(R.drawable.loading)
            .error(R.drawable.error)
            .memoryCacheKey(imageUrl)
            .diskCacheKey(imageUrl)
            .build(),
        contentDescription = contentDescription,
        contentScale = ContentScale.Crop,
        modifier = modifier
    )
}
```

### Additional Recommendations

1. **Performance**
   - Implement lazy loading for images
   - Add view model state caching
   - Include loading states for web content
   - Optimize image analysis process in AI Assistant

2. **Accessibility**
   - Add content descriptions for service icons
   - Verify color contrast with new backgrounds
   - Enhance screen reader support
   - Add proper content descriptions for decorative images

3. **Documentation**
   - Document UI constant rationale
   - Add GitHub links behavior documentation
   - Include button style migration notes
   - Document the AI Assistant feature and its limitations

## Conclusion

This PR significantly enhances the app's visual appeal and user experience through well-implemented UI improvements and new features like the AI Assistant. The changes demonstrate good attention to detail in terms of visual consistency and user interaction. While there are minor issues with test coverage, error handling, and performance optimizations, the overall changes are positive and align with good development practices.

**Recommendation**: Approve with minor changes requested for:
- Improved error handling
- Enhanced test coverage
- UI constant standardization
- Performance optimizations for image loading
- Better accessibility support
